### PR TITLE
Integration tests for contracts 

### LINKED ISSUE #26: write integration tests for contracts
Unit tests use an Externally owned account (EOAs) as the rewilder wallet. Write integration tests where `_wallet` is a 2-of-3 Gnosis Safe (https://github.com/gnosis/safe-contracts) in [RewilderDonationCampaign's constructor](https://github.com/maraoz/rewilder/blob/dapp/contracts/RewilderDonationCampaign.sol#L16), and make sure everything works properly:
- campaign can be deployed and transfered to gnosis safe,
- random accounts can donate and receive NFTs
- gnosis safe can pause campaign
- when paused, donations revert
- gnosis safe can unpause campaign
- when unpaused, donations work again and mint NFTs
- gnosis safe can finalize campaign

### PR DESCRIPTION
**Pull request by jparnaudo**
Closes #26 

Script for testing `RewilderDonationCampaign` with Gnosis Safe multisig 2of3 wallet.

To run script
- make sure you have gnosis safe-contracts deployed into your localhost node
`yarn install && npx yarn hardhat --network localhost deploy`

- run test
`npx hardhat run ./scripts/gnosis-multisig-test.js --network localhost`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,8 @@
     "@chakra-ui/react": "^1.6.3",
     "@emotion/react": "^11",
     "@emotion/styled": "^11",
+    "@gnosis.pm/safe-contracts": "^1.3.0",
+    "@material-ui/core": "^4.12.3",
```
Comment (maraoz) [inline package.json:11]: remove unneeded dependency pls

File: package.json [hunk 2]
```diff
@@ -18,7 +20,7 @@
     "@web3-react/walletconnect-connector": "^6.2.4",
     "deepmerge": "^4.2.2",
     "dotenv": "^10.0.0",
-    "ethers": "^5.4.5",
+    "ethers": "5.4.5",
```
Comment (maraoz) [inline package.json:23]: was this change intended?

**Review by maraoz — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (jparnaudo) [reply package.json:23]: Yes, see:
https://docs.ethers.io/v5/api/signer/#Signer-signTypedData

**Review by maraoz — APPROVED** [review]
LGTM, ready to merge after you include these polishing changes: https://github.com/jparnaudo/rewilder/pull/3

**Pull request merged**